ReSharper.FSharp/src/FSharp/FSharp.Psi.Intentions/src/DisableWarning/DisableWarningActionProvider.fs

```diff
@@ -0,0 +1,49 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Intentions.DisableWarning
+
+open JetBrains.Application.UI.Controls.BulbMenu.Anchors
+open JetBrains.ReSharper.Feature.Services.Bulbs
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Feature.Services.Intentions
+open JetBrains.ReSharper.Intentions.DisableWarning
+open JetBrains.ReSharper.Plugins.FSharp
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Intentions
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi
+open JetBrains.ReSharper.Psi.Files
+open JetBrains.Util
+
+[<CustomHighlightingActionProvider(typeof<FSharpProjectFileType>)>]
+type DisableWarningActionProvider(settingsManager: HighlightingSettingsManager) =
+    interface IDisableWarningActionProvider with
+        member this.GetActions(highlighting, highlightingRange, sourceFile, configureAnchor) =
+            let severityId = highlighting.GetConfigurableSeverityId()
+            if isNull severityId then EmptyList.Instance else
+
+            let hasCompilerId =
+                settingsManager.GetCompilerIds(highlighting, FSharpLanguage.Instance)
+                |> (not << Seq.isEmpty)
```

Comment:
Or it could be something like this:
```suggestion
                settingsManager.GetCompilerIds(highlighting, FSharpLanguage.Instance)
                |> Seq.isEmpty |> not
```

Choose what you like the most 🙂